pages/user.js

```diff
@@ -0,0 +1,44 @@
+import Cookies from "cookies";
+import Iron from "@hapi/iron";
+import { ENC_KEY } from "./api/users";
+import Link from "next/link";
+export async function getServerSideProps(context) {
+  const cookies = new Cookies(context.req, context.res);
+  const sessionstr = cookies.get("session");
+
+  if (sessionstr) {
+    try {
+      const session = await Iron.unseal(sessionstr, ENC_KEY, Iron.defaults);
+      if (session.loggedIn) {
+        return {
+          props: {
+            username: session.username,
+          },
+        };
+      }
+    } catch (err) {}
+  }
+  return {
+    notFound: true,
+  };
```

Comment:
This same logic could be used on any page to verify that the user is logged in. As such, it's reusable code and could benefit from being separated into it's own function.